.teamcity/src/main/kotlin/common/VersionedSettingsBranch.kt

```diff
@@ -44,6 +44,12 @@ data class VersionedSettingsBranch(val branchName: String, val enableTriggers: B
         else -> "promoteMilestone"
     }
 
+    fun promoteFinalReleaseTaskName(): String = when {
+        isMaster -> "NotSupportFinalReleaseOnMaster"
```

Comment:
If we throw an exception here, would that then cause the config to fail when we try to use it on the `master` branch? I think that would be good, so we don't need to wait for the Job to fail. WDYT?

Reply:
I thought we had same promotion job on `master` so we can't throw exception directly, but then I realized there's no final release on `master`, so yes, let's throw exception directly.